src/dataframerow/dataframerow.jl

```diff
@@ -111,15 +111,18 @@ for T in (:AbstractVector, :Regex, :Not, :Between, :All, :Colon)
         end
 
         if v isa AbstractDict
+            if all(x -> x isa AbstractString, keys(v))
+                v = (;(Symbol.(keys(v)) .=> values(v))...)
+            end
             for n in view(_names(df), idxs)
                 if !haskey(v, n)
                     throw(ArgumentError("Column :$n not found in source dictionary"))
                 end
             end
         elseif !all(((a, b),) -> a == b, zip(view(_names(df), idxs), keys(v)))
             mismatched = findall(view(_names(df), idxs) .!= collect(keys(v)))
-            throw(ArgumentError("Selected column names do not match the names in assigned value in" *
-                                " positions $(join(mismatched, ", ", " and "))"))
+            throw(ArgumentError("Selected column names do not match the names in assigned " *
+                                "value in positions $(join(mismatched, ", ", " and "))"))
         end
 
         for (col, val) in pairs(v)
```

Comment:
An alterantive way to do this would be to declare a local variable
`v_keys` which promises to be a collection of `Symbol`.
e.g. 
```
v_keys = keys(v)
v_keys = keytype(v) === Symbol ? keys(v) : Symbol.(v)
```

then each time `keys(v)` is used in this function it can be replaced with `v_keys`
The values are never used until the end

This would save having to convert the `values` as well,
and line 128 works already with symbols or strings

Reply:
this is related to the comment above - if we allow mixing strings with `Symbol`s.
Currently we do not allow mixing. When we decide on this I will rewrite it anyway.